v2/tools/generator/internal/functions/kubernetes_exporter_function.go

```diff
@@ -0,0 +1,237 @@
+
+	configMapsReference := codeGenerationContext.MustGetImportedPackageName(astmodel.GenRuntimeConfigMapsReference)
+
+	// collector := configmaps.NewCollector(<receiver>.Namespace)
+	collectorIdent := "collector"
+	var stmts []dst.Stmt //nolint:prealloc
+	stmts = append(
+		stmts,
+		astbuilder.AssignmentStatement(
+			dst.NewIdent(collectorIdent),
+			token.DEFINE,
+			astbuilder.CallQualifiedFunc(configMapsReference, "NewCollector", astbuilder.Selector(dst.NewIdent(receiverIdent), "Namespace"))))
+
+	operatorSpecSelector := astbuilder.Selector(dst.NewIdent(receiverIdent), "Spec", astmodel.OperatorSpecProperty)
+	operatorSpecConfigMapsSelector := astbuilder.Selector(operatorSpecSelector, astmodel.OperatorSpecConfigMapsProperty)
+
+	// Iterate through the mappings in alphabetical order to ensure consistent generation
+	var keys []string
+	for key := range d.mappings {
+		keys = append(keys, key)
+	}
+	sort.Strings(keys)
+
+	for _, operatorSpecPropertyName := range keys {
+		propertyPath := d.mappings[operatorSpecPropertyName]
+
+		var ifBlock dst.Stmt
+		var propertyNames []string
+		for _, prop := range propertyPath {
+			propertyNames = append(propertyNames, prop.PropertyName().String())
+		}
+
+		for i := len(propertyPath) - 1; i >= 0; i -= 1 {
+			propType := propertyPath[i].PropertyType()
+			if _, ok := astmodel.AsMapType(propType); ok {
+				panic(
+					fmt.Sprintf(
+						"Exporting Map elements as configmaps is not supported currently. Property %s has type %s",
+						propertyNames[i],
+						propType.String()))
+			}
+			if _, ok := astmodel.AsArrayType(propType); ok {
+				panic(
+					fmt.Sprintf("Exporting Slice elements as configmaps is not supported currently. Property %s has type %s",
+						propertyNames[i],
+						propType.String()))
+			}
+
+			if _, ok := astmodel.AsOptionalType(propType); !ok {
+				// If the property in question isn't actually optional, we don't need to check it so we can just continue
+				continue
+			}
+
+			var inner dst.Stmt
+			inner = ifBlock
+			if ifBlock == nil {
+				inner = d.addCollectorStmt(
+					receiverIdent,
+					collectorIdent,
+					propertyPath,
+					propertyNames,
+					operatorSpecPropertyName)
+			}
+
+			ifBlock = astbuilder.IfNotNil(
+				astbuilder.Selector(dst.NewIdent(receiverIdent), propertyNames[:i+1]...),
+				inner)
+		}
+
+		//	if <receiver>.Spec.OperatorSpec != nil && <receiver>.Spec.OperatorSpec.ConfigMaps != nil {
+		//		if <receiver>.<path> != nil {
+		//			collector.AddValue(<receiver>.Spec.OperatorSpec.ConfigMaps.<operatorSpecPropertyName>, <receiver>.<path>)
+		//		}
+		//	}
+		condition := astbuilder.JoinAnd(
+			astbuilder.NotNil(operatorSpecSelector),
+			astbuilder.NotNil(operatorSpecConfigMapsSelector))
+
+		if ifBlock == nil {
+			ifBlock = d.addCollectorStmt(
+				receiverIdent,
+				collectorIdent,
+				propertyPath,
+				propertyNames,
+				operatorSpecPropertyName)
+		}
+
+		stmts = append(
+			stmts,
+			astbuilder.SimpleIf(condition, ifBlock))
```

Comment:
Check out `astmodel.Statements()` - not only will it combine statements together, it will also flatten any `[]Stmt` you pass in.

Give your individual statements meaningful names, then call `Statements()` at the end of the sequence to stitch them together.

Reply:
Done. I still needed the list for the loop though, but renamed it to `collectStmts` (as it's the set of statements that involve the nil-guards + `collector.Add` call).